@Grover-c13 curious as I can't get it to work still - posted here:

```
def battleGym(fort):
    log.info("<>________________________________________________________________________<>")
    log.info("%s GYM NEARBY", TeamColor.DESCRIPTOR.values[fort.owned_by_team].name)
    # get gym details
    gym_proto = getGymDetails(fort)
    if not gym_proto: return 0
    log.info(gym_proto)
    log.info("<>________________________________________________________________________<>")

    ##############################################
    # gather necessary details to start a battle #
    ##############################################

    # check if gym is neutral/unowned - if so deploy to it
    if fort.owned_by_team == 0: # Neutral
        deployPokemonToGym(gym_proto)
        return 0

    # save a boolean checking if we are on the same team
    sameTeam = gl.my_team == fort.owned_by_team

    # find out who is defending
    defending_pokemon = []
    for defender in gym_proto.gym_state.memberships:
        defending_pokemon.append(defender.pokemon_data)

    # determine who we will attack with
    num = 6
    if sameTeam: num = 1
    my_attackers = pickAttackers(num) # this will be a list []

    #log.error('Function not finished, returning out...')
    #return 0

    # start the battle
    log.info('Gym ID: %s' % gym_proto.gym_state.fort_data.id)
    log.info('Attacker list: %s' % str(my_attackers))
    log.info('Defender: %d' % defending_pokemon[0].id)
    gl.api.start_gym_battle(gym_id=fort.id, attacking_pokemon_ids=my_attackers, defending_pokemon_id=defending_pokemon[0].id, player_latitude=gl.p_lat, player_longitude=gl.p_lng)

    start_reply = gl.api.call(False)
    start_payload = getPayload(start_reply)
```

Printing the request I see:

```
 [23:45:59] [ INFO] Gym ID: 555ba115a93546aca7851fed86d3ef6c.16
 [23:45:59] [ INFO] Attacker list: [17395715307789231292L] # one if my team, 6 if other
 [23:45:59] [ INFO] Defender: 568051736162843098
 [23:46:00] [ERROR] startGymBattle() request failed to get payload
```

and printing the gym_proto result:

```
 [23:45:59] [ INFO] gym_state {
  fort_data {
    id: "555ba115a93546aca7851fed86d3ef6c.16"
    last_modified_timestamp_ms: 1469762856688
    latitude: 38.854506
    longitude: -77.349488
    owned_by_team: RED
    guard_pokemon_id: FLAREON
    enabled: true
    gym_points: 2000
  }
  memberships {
    pokemon_data {
      id: 568051736162843098
      pokemon_id: FLAREON
      cp: 1002
      stamina: 130
      stamina_max: 130
      move_1: EMBER_FAST
      move_2: FLAMETHROWER
      owner_name: "Firehawk04xx"
      height_m: 0.884167253971
      weight_kg: 9.47041511536
      individual_attack: 15
      individual_defense: 14
      individual_stamina: 1
      cp_multiplier: 0.481684952974
      pokeball: ITEM_GREAT_BALL
      battles_attacked: 10
      battles_defended: 1
      creation_time_ms: 1469753995514
      num_upgrades: 2
      additional_cp_multiplier: 0.0181734859943
      nickname: "Flareon"
    }
    trainer_public_profile {
      name: "Firehawk04xx"
      level: 13
      avatar {
        skin: 2
        hair: 2
        shirt: 1
        hat: 2
        shoes: 1
        backpack: 2
      }
    }
  }
}
```

--- a/src/POGOProtos/Networking/Requests/Messages/StartGymBattleMessage.proto
+++ b/src/POGOProtos/Networking/Requests/Messages/StartGymBattleMessage.proto
@@ -4,7 +4,7 @@ package POGOProtos.Networking.Requests.Messages;
 message StartGymBattleMessage {
 	string gym_id = 1;
 	repeated fixed64 attacking_pokemon_ids = 2;
-	fixed64 defending_pokemon_id = 3;
+	fixed64 defending_pokemon_id = 3; // needs to be fixed64, tested by Grover
 	double player_latitude = 4;
 	double player_longitude = 5;
 }

